src/types/permissions.ts

```diff
@@ -32,50 +58,76 @@ export interface PermissionHandle {
   permissions: Permission[];
 }
 
-export class FungibleAssetPermission extends Serializable {
+type ToPayload<T> = { [K in keyof T]: string } & { readonly type: string };
+
+interface FungibleAssetPermissionProperties {
+  readonly asset: AccountAddress;
+  readonly amount: bigint | number;
+}
+type FungibleAssetPermissionPayload = ToPayload<FungibleAssetPermissionProperties>;
+
+export class FungibleAssetPermission extends MovePermission implements FungibleAssetPermissionProperties {
```

Comment:
So what is going on here:

Since the overall approach is to convert all permissions into a flat string, we need to first convert all inner properties to strings. This helper type (`ToPayload`) ensures we have type safety during that conversion.  

Each permission has `_PermissionProperties` which is what is stored within the class as well as `_PermissionPayload` which is used in (de)serialization. 

Fungible asset permissions are bit unique. I wanted to allow the constructor to accept amounts as `bigint` OR `number` for ease of consumption. The class converts that to a `bigint` to standardize it.